From ffcbb799ace4de3484e934fb2f661672ab634f85 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Tue, 21 Jan 2025 11:33:37 +0100
Subject: [PATCH 1581/1695] dts: stm32: tamper: use EXTI interrupt controller

Use an interrupt from EXTI hierarchical interrupt controller.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3c7bf167a170fe8b49b0996d5d9e99d8a11220a4 dts: stm32: tamper: use EXTI interrupt controller]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: Ibaa700e568bec75a3d9a369a79af555ee84afd1f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434927
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp131.dtsi | 2 +-
 core/arch/arm/dts/stm32mp211.dtsi | 2 +-
 core/arch/arm/dts/stm32mp231.dtsi | 2 +-
 core/arch/arm/dts/stm32mp251.dtsi | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index b94906ad9..b191dd230 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -455,7 +455,7 @@
 			reg = <0x5c00a000 0x400>;
 			#address-cells = <1>;
 			#size-cells = <1>;
-			interrupts = <GIC_SPI 128 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts-extended = <&exti 18 IRQ_TYPE_EDGE_RISING>;
 			clocks = <&rcc RTCAPB>;
 			ranges;
 			st,backup-zones = <10 5 17>;
diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index b0989ebde..3630073f9 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -1086,7 +1086,7 @@
 			compatible = "st,stm32mp21-tamp";
 			reg = <0x46010000 0x400>;
 			clocks = <&rcc CK_BUS_RTC>;
-			interrupts = <GIC_SPI 12 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts-extended = <&exti2 21 IRQ_TYPE_EDGE_RISING>;
 			#address-cells = <1>;
 			#size-cells = <1>;
 			ranges;
diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 772089331..7b38f5713 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -1138,7 +1138,7 @@
 			compatible = "st,stm32mp23-tamp";
 			reg = <0x46010000 0x400>;
 			clocks = <&rcc CK_BUS_RTC>;
-			interrupts = <GIC_SPI 14 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts-extended = <&exti2 21 IRQ_TYPE_EDGE_RISING>;
 			#address-cells = <1>;
 			#size-cells = <1>;
 			ranges;
diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 44340a4f4..35242bf24 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -1225,7 +1225,7 @@
 			compatible = "st,stm32mp25-tamp";
 			reg = <0x46010000 0x400>;
 			clocks = <&rcc CK_BUS_RTC>;
-			interrupts = <GIC_SPI 14 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts-extended = <&exti2 21 IRQ_TYPE_EDGE_RISING>;
 			#address-cells = <1>;
 			#size-cells = <1>;
 			ranges;
-- 
2.47.2

